File: backend/services/schemas/conteudo.py
from datetime import datetime

from models.conteudo import TipoConteudo, ApiFonte
from .base import Base


class ImagensConteudo(Base):
    capa: str | None
    banner: str | None

class ConteudoCreate(Base):
    id_externo: int
    api_fonte: ApiFonte
    tipo: TipoConteudo

class ConteudoRead(Base):
    id: int
    id_externo: int
    api_fonte: str
    tipo: TipoConteudo
    data_adicao: datetime

File: frontend/lib/types/conteudo.ts
export interface Categoria {
    id: number;
    nome: string;

}
export interface Genero {
    id: number;
    nome: string;
}

export interface Imagens {
    capa: string | null;
    banner: string | null;
}

export enum TipoConteudo {
    ANIME = "Anime",
    FILME = "Filme",
    SERIE = "Série",
}

export interface Conteudo {
    id: number;
    id_externo: number;
    api_fonte: string;
    tipo: TipoConteudo;
    data_adicao: string;
}